### ISSUE TITLE
[BUG] No iothub-connection-device-id with azure-spring-cloud-stream-binder-eventhubs

### ISSUE THREAD

**Issue by vbef-finatix** [Azure/azure-sdk-for-java#22446]
**Describe the bug**
When consuming Iot Hub messages from the built-in event hub endpoint with the spring-cloud-stream library, the `org.springframework.messaging.Message` object doesn't contain `iothub-connection-device-id` in its headers. However, while the `EventHubProcessor` is processing the message, this value is available.

***Code Snippet***
My message consumer:
```
@Bean
fun consume(): Consumer<Message<Any>> {
    return Consumer<Message<Any>> { msg ->
        println(msg)
    }
}
```

**Expected behavior**
I want to be able to read the DeviceId in a spring-cloud-stream consumer method

**Screenshots**
Here you can see that the incoming message in the `EventHubProcessor` does contain the property `iothub-connection-device-id`.
![EventData](https://user-images.githubusercontent.com/84766173/122934792-2118bf80-d370-11eb-80e5-7b0c5493e9ac.png)
But the `EventHubMessageConverter` ignores values from `systemProperties`, so that in the stream cloud consumer assignment to a DeviceId is no longer possible.
![Message Consumer](https://user-images.githubusercontent.com/84766173/122934811-24ac4680-d370-11eb-8dbb-fb635aeeda9c.png)

**Setup (please complete the following information):**
- OS: Ubunut
- IDE : IntelliJ
- Libraries: Spring Boot=2.5.1, com.microsoft.azure.sdk.iot:iot-service-client=1.32.0, com.azure.spring:azure-spring-cloud-stream-binder-eventhubs=2.5.0

**Additional context**

**Information Checklist**
Kindly make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report
- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

**Comment by yiliuTo** [comment]
@vbef-finatix , thanks for the reporting, we will fix it soon

**Comment by yiliuTo** [comment]
Hi @vbef-finatix , this bug has been fixed in pr #22683 , you could update to our latest version of 3.7.0 for the fix.
